post-processing/post_processing.py

```diff
@@ -164,6 +165,9 @@ def plot_generic(self, title, df, x_axis, y_axis, series_filters):
         for f in series_filters:
             if f[0] not in groups:
                 groups.append(f[0])
+        # all x-axis data treated as categorical
+        for g in groups:
+            df[g] = df[g].astype(str)

```

Comment:
Might be a bit controversial if we want to reuse this filtered dataframe elsewhere for another graph type, but we could always get pandas to re-infer numeric types if necessary. As it stands, Bokeh needs its bar chart categories to be passed as strings.

Reply:
This is not a problem for now, but when the numeric/scaling plotting function is created in the future, you can have a line like this for the x-axis column, that converts it to numeric.